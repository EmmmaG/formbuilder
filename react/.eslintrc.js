module.exports = {
    globals: {},
    env: {
        commonjs: true,
        es6: true,
    },
    parser: '@typescript-eslint/parser',
    plugins: ['@typescript-eslint'],
    parserOptions: {
        ecmaVersion: 2018,
        ecmaFeatures: {
            globalReturn: false,
        },
        sourceType: 'module',
    },
}
