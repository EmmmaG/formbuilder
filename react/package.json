{
  "name": "hcard-builder",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@material-ui/core": "^3.9.2",
    "@material-ui/icons": "^3.0.2",
    "@types/react-dom": "^16.8.3",
    "react": "^16.8.4",
    "react-dom": "^16.8.4",
    "react-scripts": "2.1.8",
    "typescript": "^3.3.4000"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "jest",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": [
    ">0.2%",
    "not dead",
    "not ie <= 11",
    "not op_mini all"
  ],
  "description": "This project was to conver a fixed format file into csv",
  "main": "index.js",
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@typescript-eslint/eslint-plugin": "^1.5.0",
    "@typescript-eslint/parser": "^1.5.0",
    "@types/enzyme": "^3.1.11",
    "@types/enzyme-adapter-react-16": "^1.0.1",
    "babel-core": "^6.26.3",
    "babel-loader": "^7.1.5",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-es2015-modules-commonjs": "^6.26.2",
    "babel-plugin-transform-object-rest-spread": "^6.23.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "babel-runtime": "^6.25.0",
    "enzyme": "^3.2.0",
    "enzyme-adapter-react-16": "^1.1.1",
    "enzyme-to-json": "^3.3.4",
    "jest": "^24.1.0",
    "jest-enzyme": "^7.0.1",
    "jest-fetch-mock": "^1.6.5",
    "ts-jest": "^22.4.6"
  },
  "author": "",
  "license": "ISC"
}
