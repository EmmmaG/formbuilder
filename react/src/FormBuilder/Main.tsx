import * as React from 'react'
import { withStyles, WithStyles } from '@material-ui/core/styles'

import FormBuilder, { Field } from './FormBuilder'
import FormRenderer from './FormRenderer'

const styles = () => ({
    container: {
        display: 'flex',
        flexDirection: 'row' as 'row',
        justifyContent: 'space-evenly',
        flexWrap: 'wrap' as 'wrap',
    },
})

export interface Props extends WithStyles<typeof styles> {
}

export interface State {
    fields: Field[]
	isLoading: boolean
}

export class Main extends React.Component<Props, State> {
    public constructor(props: Props) {
        super(props)

        this.state = {
            fields: [],
			isLoading: false,
        }
    }

    private onFieldsCreate = (field: Field) => {
        const currentFields = this.state.fields
        const newField = {
            ...field,
            id: currentFields.length
                ? currentFields[currentFields.length - 1].id! + 1
                : 1
        }
        currentFields.push(newField)

        this.setState({ fields: currentFields })
    }

    private onFieldDelete = (id: number) => {
        const currentFields = this.state.fields
        const newFields = currentFields.filter(f => f.id !== id)

        this.setState({ fields: newFields })
    }

    private fetchData = async() => {
		this.setState({ isLoading: true})

		await fetch('http://localhost:8001/fields')
			.then(response => {
				return response.json()
			}).then(data => {
				this.setState({ fields: data })
			})

		this.setState({ isLoading: false })
	}


    public render() {
        const { classes } = this.props
        const { fields, isLoading } = this.state

        return (
            <div className={classes.container}>
                <FormBuilder onFieldsCreate={this.onFieldsCreate} fetchData={this.fetchData}/>
                <FormRenderer onDelete={this.onFieldDelete} fields={fields} />
            </div>
        )
    }
}

export default withStyles(styles)(Main)
