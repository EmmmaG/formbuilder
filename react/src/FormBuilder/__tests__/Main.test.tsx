import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import Main, { Props } from '../Main'

const props: Props = {
	classes: idObj,
}

describe('Main', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<Main {...props} />).dive()
		expect(wrapper).toMatchSnapshot()
	})
})
