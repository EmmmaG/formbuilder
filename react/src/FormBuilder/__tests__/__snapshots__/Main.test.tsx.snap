// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
<div
  className="Main-container-1"
>
  <WithStyles(FormBuilder)
    fetchData={[Function]}
    onFieldsCreate={[Function]}
  />
  <WithStyles(FormRenderer)
    fields={Array []}
    onDelete={[Function]}
  />
</div>
`;
